img_processing/src/distance.cpp

```
@@ -0,0 +1,126 @@
+{
+    if (!instance) {
+        if (image.empty())
+            Manager::imgLogger.logMessage(logger::LogLevel::ERROR,
+                                          "No image was provided");
+        else
+            instance = new Distance(image);
+    }
+    return *instance;
+}
+
+void Distance::setFocalLength(double focalLength)
+{
+    this->focalLength = focalLength;
+}
+
+// function that finds the distance of the objects and update them in
+// detectionObjects
+void Distance::findDistance(std::vector<ObjectInformation> &objectInformations)
+{
+    int knownSize, imageSize;
+    // Move over all objects
+    for (ObjectInformation &objectInformation : objectInformations) {
+        // is person
+        if (objectInformation.type == 0) {
+            // distance test based on the position of the legs in the image
+            if (objectInformation.position.y +
+                    objectInformation.position.height >
+                MIN_LEGAL_HEIGHT) {
+                objectInformation.distance = 0;
+                return;
+            }
+            // Find the size of the object in reality and in the picture
+            knownSize = PERSON_HEIGHT;
+            imageSize = objectInformation.position.height;
+        }
+        else {
+            // Find the size of the object in reality and in the picture
+            knownSize = CAR_WIDTH;
+            imageSize = objectInformation.position.width;
+        }
+        // Calculate the distance in meters
+        objectInformation.distance =
+            (focalLength * knownSize / imageSize) / 1000;
+    }
+}
+
+void Distance::findFocalLength(const cv::Mat &image)
+{
+    // Check if the input image is empty
+    if (image.empty()) {
+        Manager::imgLogger.logMessage(logger::LogLevel::ERROR,
+                                      "Could not open or find the image");
+        return;
+        // throw std::runtime_error("Could not open or find the image");
```

Is it ok to return, or we do need to throw an error?